packages/cli/src/databases/config.ts

```diff
@@ -47,16 +48,23 @@ export const getOptionOverrides = (dbType: 'postgresdb' | 'mysqldb') => ({
 	password: config.getEnv(`database.${dbType}.password`),
 });
 
-const getSqliteConnectionOptions = (): SqliteConnectionOptions => ({
-	type: 'sqlite',
-	...getCommonOptions(),
-	database: path.resolve(
-		Container.get(InstanceSettings).n8nFolder,
-		config.getEnv('database.sqlite.database'),
-	),
-	enableWAL: config.getEnv('database.sqlite.enableWAL'),
-	migrations: sqliteMigrations,
-});
+const getSqliteConnectionOptions = (): SqliteConnectionOptions | SqlitePooledConnectionOptions => {
+	const poolSize = config.getEnv('database.sqlite.poolSize');
+	const commonOptions = {
+		...getCommonOptions(),
+		database: path.resolve(
+			Container.get(InstanceSettings).n8nFolder,
+			config.getEnv('database.sqlite.database'),
+		),
+		enableWAL: config.getEnv('database.sqlite.enableWAL'),
+		migrations: sqliteMigrations,
+	};
+	if (poolSize > 0) {
+		return { type: 'sqlite-pooled', poolSize, ...commonOptions };
+	} else {
+		return { type: 'sqlite', ...commonOptions };
+	}
```

Comment:
I was thinking we could have done this condition within the library, in the `DriverFactory`, but no objections to doing it here either

Reply:
I want it to do it that way eventually as well, but for now, I want to get this moving without making any changes to the default sqlite driver. Once pooling seems mature enough, maybe we can just delete the old driver.